Sources/ForageSDK/Foundation/Network/LiveForageService.swift

```diff
@@ -133,6 +134,67 @@ class LiveForageService: ForageService {
         let sessionToken = ForageSDK.shared.sessionToken
         let merchantID = ForageSDK.shared.merchantID
 
+        do {
+            let (vaultResponse, forageRequest) = try await collectPinForPayment(
+                pinCollector: pinCollector,
+                paymentReference: paymentReference,
+                action: .capturePayment
+            )
+
+            _ = try await awaitResult { completion in
+                self.pollingService.execute(
+                    vaultResponse: vaultResponse,
+                    request: forageRequest,
+                    completion: completion
+                )
+            }
+
+            return try await awaitResult { completion in
+                self.getPayment(
+                    sessionToken: sessionToken,
+                    merchantID: merchantID,
+                    paymentRef: paymentReference,
+                    completion: completion
+                )
+            }
+        } catch {
+            throw error
+        }
+    }
+
+    func getPayment(sessionToken: String, merchantID: String, paymentRef: String, completion: @escaping (Result<PaymentModel, Error>) -> Void) {
+        do { try provider.execute(model: PaymentModel.self, endpoint: ForageAPI.getPayment(sessionToken: sessionToken, merchantID: merchantID, paymentRef: paymentRef), completion: completion) } catch { completion(.failure(error)) }
+    }
+
+    // MARK: Collect PIN
+
+    func collectPinForDeferredCapture(
+        pinCollector: VaultCollector,
+        paymentReference: String
+    ) async throws -> VaultResponse {
+        do {
+            let collectPinResult = try await collectPinForPayment(
+                pinCollector: pinCollector,
+                paymentReference: paymentReference,
+                action: .collectPin
```

Comment:
Adjust to `.deferCapture`.

Reply:
c65c391da582315b639a0159fe41487bf5b24e3d